sqlalchemy_ibmi/base.py

```diff
@@ -259,7 +259,11 @@ class DB2TypeCompiler(compiler.GenericTypeCompiler):
     """IBM i Db2 Type Compiler"""
 
     def visit_TIMESTAMP(self, type_, **kw):
-        return "TIMESTAMP"
+        return "TIMESTAMP%s" % (
+            "(%d)" % type_.precision
+            if getattr(type_, "precision", None) is not None
+            else ""
+        )
```

Comment:
We don't support Python < 3.6 so I'd rather use f-strings than percent formatting (which was already worse than str.format).
 
```suggestion
        precision = getattr(type_, "precision", None)
        if precision is not None:
            return f"TIMESTAMP({precision})"
        else:
            return "TIMESTAMP
```

Reply:
Alternatively, `precision = getattr(type_, "precision", 6)` and we don't need the if check at all.